### PR TITLE
[ONNX] Enable export of __xor_ (#64042)

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#64581 [ONNX] Enable export of __xor_ (#64042)**
* #64580 [ONNX] Fix inplace fill_ dtype export mismatch (#64233)
* #64579 [ONNX] Update doc and error message for indexing export (#64290)
* #64578 [ONNX] Fix remainder export (#64230)
* #64381 Add onnx test for batched_nms (#53175)
* #64379 [ONNX] Fix gather squeeze axis in constant folding (#63588)
* #64378 [ONNX] Fix cuda test case (#63597)
* #64377 [ONNX] Fix input sequence for pad op (#60554)
* #64376 Fix empty size constant creation (#63607)
* #64375 [ONNX] Update instance_norm implementation and support training (#60538)

* Enbale xor

* Update test_pytorch_onnx_onnxruntime.py

* Update symbolic_opset9.py

* Update symbolic_opset9.py

* Update test_pytorch_onnx_onnxruntime.py

* Update symbolic_opset9.py

Differential Revision: [D30919598](https://our.internmc.facebook.com/intern/diff/D30919598)

### CODE REVIEW

**Comment by msaroufim** [comment]
@msaroufim has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30919598).

File: torch/onnx/symbolic_opset9.py [hunk 1]
```diff
@@ -1084,10 +1084,11 @@ def __interpolate(g, input, size, scale_factor, mode , align_corners, recompute_
                                                              mode , align_corners)
     return g.op("Upsample", input, scales, mode_s=mode)
 
-@parse_args("v")
+
 def bitwise_not(g, inp):
     if inp.type().scalarType() != "Bool":
-        return _unimplemented("bitwise_not", "non-bool tensor")
+        raise NotImplementedError("ONNX export does NOT support exporting bitwise Not " +
+                                  "for non-boolean input values")
```
Comment (malfet) [inline torch/onnx/symbolic_opset9.py:1091]: String literals are automatically concatenated https://docs.python.org/3/reference/lexical_analysis.html#string-literal-concatenation
```suggestion
        raise NotImplementedError("ONNX export does NOT support exporting bitwise Not "
                                  "for non-boolean input values")
```

File: torch/onnx/symbolic_opset9.py [hunk 2]
```diff
@@ -1116,6 +1117,9 @@ def wrap_with_not(g, input, other):
 
 
 def __not_(g, self):
+    if self.type().scalarType() != "Bool":
+        raise NotImplementedError("ONNX export does NOT support exporting bitwise Not " +
+                                  "for non-boolean input values")
```
Comment (malfet) [inline torch/onnx/symbolic_opset9.py:1122]: Same as above
```suggestion
        raise NotImplementedError("ONNX export does NOT support exporting bitwise Not "
                                  "for non-boolean input values")
```

File: torch/onnx/symbolic_opset9.py [hunk 3]
```diff
@@ -1162,14 +1166,31 @@ def le(g, input, other):
...
 
-@wrap_logical_op_with_cast_to_and_from("Bool")
 def __and_(g, input, other):
-    return g.op("And", input, other)
+    if input.type().scalarType() == "Bool" and \
+            other.type().scalarType() == "Bool":
+        return g.op("And", input, other)
+    else:
+        raise NotImplementedError("ONNX export does NOT support exporting bitwise AND " +
+                                  "for non-boolean input values")
```
Comment (malfet) [inline torch/onnx/symbolic_opset9.py:1175]: Same as above
```suggestion
        raise NotImplementedError("ONNX export does NOT support exporting bitwise AND "
                                  "for non-boolean input values")
```

File: torch/onnx/symbolic_opset9.py [hunk 4]
```diff
@@ -1162,14 +1166,31 @@ def le(g, input, other):
...
 
-@wrap_logical_op_with_cast_to_and_from("Bool")
 def __or_(g, input, other):
-    return g.op("Or", input, other)
+    if input.type().scalarType() == "Bool" and \
+            other.type().scalarType() == "Bool":
+        return g.op("Or", input, other)
+    else:
+        raise NotImplementedError("ONNX export does NOT support exporting bitwise OR " +
+                                  "for non-boolean input values")
```
Comment (malfet) [inline torch/onnx/symbolic_opset9.py:1184]: Same as above
```suggestion
        raise NotImplementedError("ONNX export does NOT support exporting bitwise OR "
                                  "for non-boolean input values")
```

File: torch/onnx/symbolic_opset9.py [hunk 5]
```diff
@@ -1162,14 +1166,31 @@ def le(g, input, other):
...
+
+
+def __xor_(g, input, other):
+    if input.type().scalarType() == "Bool" and \
+            other.type().scalarType() == "Bool":
+        return g.op("Xor", input, other)
+    else:
+        raise NotImplementedError("ONNX export does NOT support exporting bitwise XOR " +
+                                  "for non-boolean input values")
```
Comment (malfet) [inline torch/onnx/symbolic_opset9.py:1193]: Same as above
```suggestion
        raise NotImplementedError("ONNX export does NOT support exporting bitwise XOR "
                                  "for non-boolean input values")
```

**Review by malfet — APPROVED** [review]

**Comment by msaroufim** [comment]
@msaroufim has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30919598).

**Comment by malfet** [comment]
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30919598).

**Comment by BowenBao** [comment]
@malfet rebased, please import.

**Comment by malfet** [comment]
@BowenBao Not going to reimport in that case

**Pull request closed without merging**